project/src/pages/room/room.tsx

```diff
@@ -35,8 +40,19 @@ function Room({ offers, reviews, cities }: RoomProps): JSX.Element {
     window.scroll(0, 0);
   };
 
+  useEffect(() => {
+    store.dispatch(fetchReviewsAction(Number(id)));
+    store.dispatch(fetchNearbyOffersAction(Number(id)));
+  }, [id]);
+
+  useEffect(() => {
+    if(!offer) {
+      store.dispatch(fetchOfferAction(Number(id)));
+    }
+  }, [offer, id]);
+
 
```

Comment:
сейчас при переходе на страницу оффера я вижу

1. сначала 404
2. затем страничку оффера

хотелось бы сначала видеть состояние загрузки, затем страничку оффера

```suggestion
if (isLoading) {
  return <LoadingSpinner />;
}
```